src/application/tester/tester.go

```diff
@@ -52,17 +58,13 @@ func TestApp(appContext types.AppContext, writer io.Writer, mode composebuilder.
 
 // TestService runs the tests for the service and return true if the tests passed
 // and an error if any
-func TestService(serviceContext types.ServiceContext, writer io.Writer, mode composebuilder.BuildMode, shutdown chan os.Signal) (types.TestResult, error) {
-	util.PrintSectionHeaderf(writer, "Testing service '%s'\n", serviceContext.Dir)
-	serviceTester, err := NewServiceTester(serviceContext, writer, mode)
-	if err != nil {
-		return types.TestResult{}, err
-	}
+func TestService(testRunner *TestRunner, testRole string, writer io.Writer, shutdown chan os.Signal) (types.TestResult, error) {
```

Comment:
Can we keep TestService with the api it had and create a private `runServiceTest` with the shared implementation? I want to keep as much logic out of the cmd folder as we can